docs/api/QUIC_STREAM_EVENT.md

```diff
@@ -117,11 +117,13 @@ Absolute offset of the current data payload from the start of the receive operat
 
 MsQuic indicates the total buffer length of the data in this parameter.
 
-If the application can processes only part of the received buffer synchronously in this call, it can be indicated to MsQuic by setting this parameter to the byte count processed and returning `QUIC_STATUS_CONTINUE` from this call.
-If the application desires to process the received data asynchronously, it should return `QUIC_STATUS_PENDING` from this call.
-
 See [Receiving Data On Streams](../Streams.md#Receiving) for further details on receiving data on a stream.
 
+Most notably, if the application...
+ - processes all the received data synchronously in this call, this parameter must be left unchanged and  `QUIC_STATUS_SUCCESS` must be returned from this call.
+ - can processes only part of the received buffer synchronously in this call, it should be indicated to MsQuic by setting this parameter to the byte count processed and returning `QUIC_STATUS_CONTINUE` from this call.
```

Comment:
```suggestion
 - can process only part of the received buffer synchronously in this call, it should be indicated to MsQuic by setting this parameter to the byte count processed and returning `QUIC_STATUS_CONTINUE` from this call.
```

Reply:
I think it should be more forceful than "...it should be indicated to MsQuic...". If only part of the data is processed, the app MUST change the TotalBufferLength, otherwise they'll lose data that wasn't processed